src/utils/wallet.js

```diff
@@ -690,37 +678,26 @@ class Wallet {
         }
 
         const tempKeyStore = new nearApiJs.keyStores.InMemoryKeyStore()
-
-        const connection = nearApiJs.Connection.fromConfig({
+        this.connection = nearApiJs.Connection.fromConfig({
             networkId: NETWORK_ID,
             provider: { type: 'JsonRpcProvider', args: { url: NODE_URL + '/' } },
             signer: new nearApiJs.InMemorySigner(tempKeyStore)
         })
         await Promise.all(accountIds.map(async (accountId, i) => {
-            const account = new nearApiJs.Account(connection, accountId)
             this.accountId = accountId
+            await this.loadAccountAndState()
+            if (fromSeedPhraseRecovery) {
+                this.has2fa = false
```

Comment:
hmm, if account isn't recovered from seed phrase it doesn't mean it has 2FA. So I am not sure what is the logic here.

Reply:
The account we're loading is a temporary account based on the recovery private key material.

Since it is a seed phrase it is going to be a FAK so we can skip 2FA modal confirm when adding the new localStorage key.

Once the new key is added, the actual account will be loaded and if it has 2FA enabled, it will be a 2FA account.